### PR TITLE
BUGFIX: SD-540 handle null value in stacked chart

### PR DESCRIPTION
**Pull request by naibinh**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Squash commits

# Related PRs
- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/2343
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2108

# Related Jira tasks
- SD-540: https://jira.intgdc.com/browse/SD-540

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4199/ : SUCCESS in 12m 06s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5344/ : SUCCESS in 10s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2266/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1634/ : UNSTABLE in 13m 33s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5345/ : SUCCESS in 18s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4200/ : SUCCESS in 5m 38s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5346/ : SUCCESS in 11s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2267/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1635/ : SUCCESS in 13m 21s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5347/ : SUCCESS in 20s

**Comment by naibinh** [comment]
extended test

**Comment by yenkins** [comment]
Demo is running at https://client-demo.na.intgdc.com:50029/storybook/

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1636/ : FAILURE in 8m 11s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5348/ : SUCCESS in 9s

**Comment by naibinh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1065.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

File: src/components/visualizations/chart/highcharts/test/getZeroAlignConfiguration.spec.ts [hunk 1]
```diff
@@ -205,6 +209,171 @@ describe("getZeroAlignConfiguration", () => {
         });
     });
 
+    describe("getMinMaxInfo with some null data", () => {
+        const SERIES: ISeriesItem[] = [
+            {
+                yAxis: 0,
+                data: [null, null].map((num: number) => ({ y: num })),
```
Comment (ptquang86) [inline src/components/visualizations/chart/highcharts/test/getZeroAlignConfiguration.spec.ts]: Could we re-use this function?
and change `num: number` => `num: number | null`

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/1640/ : SUCCESS in 34m 02s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5353/ : SUCCESS in 12s

File: src/components/visualizations/chart/highcharts/test/getZeroAlignConfiguration.spec.ts [hunk 2]
```diff
@@ -205,6 +209,171 @@ describe("getZeroAlignConfiguration", () => {
...
+                    id: 1,
+                    min: 0,
+                    max: 500,
+                    isSetMin: false,
+                    isSetMax: false,
+                },
+            ]);
+        });
+
+        it("should return min/max in total with percent stack chart", () => {
```
Comment (ptquang86) [inline src/components/visualizations/chart/highcharts/test/getZeroAlignConfiguration.spec.ts]: Should we use data driven test for this 3 tests? (line 232, 256 and 280)

Comment (naibinh) [reply src/components/visualizations/chart/highcharts/test/getZeroAlignConfiguration.spec.ts]: done

File: src/components/visualizations/chart/highcharts/test/getZeroAlignConfiguration.spec.ts (hunk 1 above)
Comment (naibinh) [reply src/components/visualizations/chart/highcharts/test/getZeroAlignConfiguration.spec.ts]: done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4205/ : SUCCESS in 4m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5360/ : SUCCESS in 6s

**Comment by ptquang86** [comment]
LGTM

File: src/components/visualizations/chart/highcharts/test/getZeroAlignConfiguration.spec.ts [hunk 3]
```diff
@@ -205,31 +211,155 @@ describe("getZeroAlignConfiguration", () => {
         });
     });
 
+    describe("getMinMaxInfo with some null data", () => {
```
Comment (ivan-nejezchleb) [inline src/components/visualizations/chart/highcharts/test/getZeroAlignConfiguration.spec.ts]: should not this be as child describe inside `describe("getMinMaxInfo", () => {`?

Comment (naibinh) [reply src/components/visualizations/chart/highcharts/test/getZeroAlignConfiguration.spec.ts]: done, please check my latest commit, @ivan-nejezchleb

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4226/ : SUCCESS in 4m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/5381/ : SUCCESS in 12s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/2279/ : SUCCESS in 16m 05s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/4228/ : SUCCESS in 4m 59s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1084/ : SUCCESS in 27s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/1305/ : SUCCESS in 28m 11s

**Pull request merged**